youtube_dl/extractor/tumblr.py

```diff
@@ -68,8 +99,35 @@ def _real_extract(self, url):
 
         iframe = self._download_webpage(iframe_url, video_id,
                                         'Downloading iframe page')
-        video_url = self._search_regex(r'<source src="([^"]+)"',
-                                       iframe, 'video url')
+
+        sd_video_url = self._search_regex(r'<source src="([^"]+)"',
+                                          iframe, 'sd video url')
+        format_id = sd_video_url.split("/")[-1] + 'p'
+        if len(format_id) != 4:
+            format_id = "480p"
+        video_urls.append({
+            'ext': 'mp4',
+            'format_id': 'sd',
+            'url': sd_video_url,
+            'resolution': format_id,
+        })
+
+        try:
+            hd_video_url = self._search_regex(r'hdUrl":"([^"]+)"',
+                                              iframe, 'hd video url')
+            hd_video_url = hd_video_url.replace("\\", "")
+            format_id = hd_video_url.split("/")[-1] + 'p'
+            if len(format_id) != 4:
```

Comment:
This may not be a good check. ```format_id``` like 'abcp' passes the check, too. On the other hand, we usually leave the resolution field ```None``` if it's unknown.